lib/srv/db/proxyserver.go

```diff
@@ -270,6 +280,50 @@ func (s *ProxyServer) Proxy(ctx context.Context, clientConn, serviceConn io.Read
 	return trace.NewAggregate(errs...)
 }
 
+func (s *ProxyServer) monitorConn(ctx context.Context, conn net.Conn) (net.Conn, error) {
+	authContext, err := s.cfg.Authorizer.Authorize(ctx)
```

Comment:
`Authorize` is already done in `Connect` above - I would probably rather modify it to return authContext as well and then pass it here from `(postgres|mysql)/proxy.go`.